protocol/error.go

```diff
@@ -1,10 +1,15 @@
 // Defines constants representing the types
-// of errors that the server may return to a client.
+// of errors that the server may return to a client,
+// and that the client may throw after a consistency
+// check or cryptographic verification
 
 package protocol
 
+// An ErrorCode is a special int.
```

Comment:
The most interesting fact about `ErrorCode` isn't that it is internally an `int`, but rather that it is a integer representation of (specific) error-messages (those that travel through the wire??) 
Also, it's interesting (for an overview) that they can be mapped to human readable string representations.